cylc/flow/cfgspec/suite.py

```diff
@@ -1604,12 +1606,27 @@ def upgrade_param_env_templates(cfg, descr):
 
 
 def warn_about_depr_platform(cfg):
-    """Warn if deprecated host or batch system appear in config."""
-    if 'runtime' in cfg:
-        for task_name, task_cfg in cfg['runtime'].items():
-            platform = get_platform(task_cfg, task_name, warn_only=True)
-            if type(platform) == str:
-                LOG.warning(platform)
+    """Validate platforms config.
+
+    - Warn if deprecated host or batch system appear in config,
+    - Raise if platforms section is also present.
```

Comment:
You know that the reason that this wasn't here is that not all instances where this conflict happens will appear here? Some conflicts don't appear until after task inheritance is expanded.